### DISCUSSION TITLE
Support for dynamic JSX tags

### DISCUSSION THREAD

**Discussion by norbertgogibiro** [solidjs/solid#1027] [Q&A] [answered]
I would like to set JSX tag names dynamically in SolidJS. I come from React where [it is fairly simple to do](https://stackoverflow.com/questions/33471880/dynamic-tag-name-in-react-jsx):

```
/* Working ReactJS Code: */
export default MyWrapper = ({ children, ..attributes }) => {
  const Element = "div";

  return (
    <Element {...attributes}>
      {children}
    </Element>
  )
}
```

but when I try to do the same thing in SolidJS, I get the following error:
```
/* Console output when trying to do the same in SolidJS: */
dev.js:530 Uncaught (in promise) TypeError: Comp is not a function
  at dev.js:530:12
  at untrack (dev.js:436:12)
  at Object.fn (dev.js:526:37)
  at runComputation (dev.js:706:22)
  at updateComputation (dev.js:691:3)
  at devComponent (dev.js:537:3)
  at createComponent (dev.js:1236:10)
  at get children [as children] (Input.jsx:38:5)
  at _Hot$$Label (Input.jsx:7:24)
  at @solid-refresh:10:42
```

I would like to know if I miss something here, or whether it is possible to achieve this in SolidJS in any other way.

I asked the same question on StackOverflow too:
https://stackoverflow.com/questions/72443438/dynamic-tag-name-in-solid-jsx

**Comment by atk** [comment, marked as answer]
And the answer on StackOverflow is correct, too. In Solid, you use the <Dynamic> component for dynamic tags. While Solid could theoretically support such a pattern, it would easily lead to confusion, as you can no longer instantly see if "Element" is an intrinsic or a functional component.
